pkl-binary-java/build.gradle.kts

```
@@ -0,0 +1,54 @@
+plugins {
+    // apply the Pkl plugin
+    alias(libs.plugins.pkl)
+    idea
+    `java-library`
+}
+
+java {
+    sourceCompatibility = JavaVersion.VERSION_17
+}
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    implementation(libs.pklConfigJavaAll)
+}
+
+// Generate a resource named "data.msgpack" by evaluating data.pkl
+pkl {
+    evaluators {
+        register("configData") {
+            sourceModules.set(files("data.pkl"))
+            outputFormat.set("pkl-binary")
+            outputFile.set(file("build/generated/pkl-binary/configData/data.msgpack"))
```

A common pattern in Java is to put the resource under the same path as the class that loads it.

So your classpath looks like:

```
example/JavaPklBinaryExample.class
example/data.msgpack
```

```suggestion
            outputFile.set(file("build/generated/pkl-binary/example/data.msgpack"))
```